### PR TITLE
Add Commands for fsc and fsi

### PR DESCRIPTION
**Pull request by KevinRansom**
This adds dotnet fsc and dotnet fsi commands to dotnet.exe
This is intendended for VS2016 RTM, is this the correct branch?

Still needs additional test cases.

````
c:\kevinransom\cli>dotnet /?
.NET Command Line Tools (2.2.200-preview-009431)
Usage: dotnet [runtime-options] [path-to-application] [arguments]

Execute a .NET Core application.

runtime-options:
  --additionalprobingpath <path>     Path containing probing policy and assemblies to probe for.
  --additional-deps <path>           Path to additional deps.json file.
  --fx-version <version>             Version of the installed Shared Framework to use to run the application.
  --roll-forward-on-no-candidate-fx  Roll forward on no candidate shared framework is enabled.

path-to-application:
  The path to an application .dll file to execute.

Usage: dotnet [sdk-options] [command] [command-options] [arguments]

Execute a .NET Core SDK command.

sdk-options:
  -d|--diagnostics  Enable diagnostic output.
  -h|--help         Show command line help.
  --info            Display .NET Core information.
  --list-runtimes   Display the installed runtimes.
  --list-sdks       Display the installed SDKs.
  --version         Display .NET Core SDK version in use.

SDK commands:
  add               Add a package or reference to a .NET project.
  build             Build a .NET project.

   …

Additional commands from bundled tools:
  dev-certs         Create and manage development certificates.
  ef                Entity Framework Core command-line tools.
  fsc               Execute F# compiler.
  fsi               Start F# Interective / execute F# scripts.
  sql-cache         SQL Server cache command-line tools.
  user-secrets      Manage development user secrets.
  watch             Start a file watcher that runs a command when files change.

Run 'dotnet [command] --help' for more information on a command.

````

### CODE REVIEW

**Comment by livarcocc** [comment]
You may want to retarget this PR to release/2.1.6xx, which will be the CLI version included by default in Dev16.

**Comment by KevinRansom** [comment]
@livarcocc thanks that is good info.

**Comment by wli3** [comment]
@dotnet-bot test opensuse.42.3 x64 Debug Build

File: src/dotnet/Telemetry/TelemetryFilter.cs [hunk 1]
```diff
@@ -86,6 +86,16 @@ public IEnumerable<ApplicationInsightsEntryFormat> Filter(object objectToFilter)
                 optionsToLog: new HashSet<string> {"sdk-package-version"}
             ),
             new TopLevelCommandNameAndOptionToLog
+            (
```
Comment (wli3) [inline src/dotnet/Telemetry/TelemetryFilter.cs]: This is not needed. If you only need toplevel command name.

**Comment by wli3** [comment]
@dotnet-bot test RHEL7.2 x64 Release Build

**Comment by wli3** [comment]
@dotnet-bot test Ubuntu16.04 x64 Debug Build

File: scripts/obtain/dotnet-install.ps1 [hunk 2]
```diff
@@ -219,7 +219,7 @@ function GetHTTPResponse([Uri] $Uri)
             }
             # Default timeout for HttpClient is 100s.  For a 50 MB download this assumes 500 KB/s average, any less will time out
             # 10 minutes allows it to work over much slower connections.
-            $HttpClient.Timeout = New-TimeSpan -Minutes 10
```
Comment (wli3) [inline scripts/obtain/dotnet-install.ps1]: why this change?

**Comment by wli3** [comment]
the code looks good so far, you could merge if you think it is ready

**Comment by KevinRansom** [comment]
@wli3 ,  I do … thanks
